lib/mayaUsd/fileio/primUpdaterManager.h

```diff
@@ -103,14 +153,64 @@ class PrimUpdaterManager : public PXR_NS::TfWeakBase
     //! Create the pull parent and set it into the prim updater context.
     MDagPath setupPullParent(const Ufe::Path& pulledPath, VtDictionary& args);
 
+    //! Record pull information for the pulled path, for inspection on
+    //! scene changes.
+    void recordPullVariantInfo(const Ufe::Path& pulledPath, const MDagPath& pullParentPath);
+
+    // Maya file new or open callback.  Member function to access other private
+    // member functions.
+    static void beforeNewOrOpenCallback(void* clientData);
+
+    void beginManagePulledPrims();
```

Comment:
Start listening to scene structure changes when we pull, to know when to hide or show a pulled hierarchy.